src/main/java/com/fasterxml/jackson/core/JsonGenerator.java

```diff
@@ -2491,13 +2491,17 @@ public void copyCurrentEvent(JsonParser p) throws IOException
         }
         case ID_NUMBER_FLOAT:
         {
-            NumberType n = p.getNumberType();
-            if (n == NumberType.BIG_DECIMAL) {
-                writeNumber(p.getDecimalValue());
-            } else if (n == NumberType.FLOAT) {
-                writeNumber(p.getFloatValue());
+            Number n = p.getNumberValueExact();
+            if (n instanceof Float) {
+                writeNumber((Float) n);
+            } else if (n instanceof BigInteger) {
```

Comment:
Never happens, it's floating point numbers.

Reply:
I've now removed this BigInteger check